### ISSUE TITLE
Issue on install

### ISSUE THREAD

**Issue by MatthewKennedy** [BrendanThompson/rb-scpt#22]
```bash
gem install rb-scpt
Fetching rb-scpt-1.0.3.gem
Building native extensions. This could take a while...
ERROR:  Error installing rb-scpt:
	ERROR: Failed to build gem native extension.

    current directory: /Users/matthewkennedy/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/rb-scpt-1.0.3
/Users/matthewkennedy/.asdf/installs/ruby/3.3.0/bin/ruby extconf.rb
create /Users/matthewkennedy/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/rb-scpt-1.0.3/src/osx_ruby.h ...
create /Users/matthewkennedy/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/rb-scpt-1.0.3/src/osx_intern.h ...
creating Makefile

current directory: /Users/matthewkennedy/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/rb-scpt-1.0.3
make DESTDIR\= sitearchdir\=./.gem.20240412-60125-dfri47 sitelibdir\=./.gem.20240412-60125-dfri47 clean

current directory: /Users/matthewkennedy/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/rb-scpt-1.0.3
make DESTDIR\= sitearchdir\=./.gem.20240412-60125-dfri47 sitelibdir\=./.gem.20240412-60125-dfri47
compiling src/SendThreadSafe.c
src/SendThreadSafe.c:362:16: warning: converting the enum constant to a boolean [-Wint-in-bool-context]
  if (sendMode && kAEWaitReply) {
               ^
1 warning generated.
compiling src/rbae.c
src/rbae.c:62:1: warning: function 'rbAE_raiseMacOSError' could be declared with attribute 'noreturn' [-Wmissing-noreturn]
{
^
src/rbae.c:186:35: warning: implicit conversion loses integer precision: 'long' to 'AETransactionID' (aka 'int') [-Wshorten-64-to-32]
  AETransactionID transactionID = NUM2LONG(transactionIDValue);
                  ~~~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/matthewkennedy/.asdf/installs/ruby/3.3.0/include/ruby-3.3.0/ruby/internal/arithmetic/long.h:51:22: note: expanded from macro 'NUM2LONG'
#define NUM2LONG     RB_NUM2LONG          /**< @old{RB_NUM2LONG} */
                     ^
/Users/matthewkennedy/.asdf/installs/ruby/3.3.0/include/ruby-3.3.0/ruby/internal/arithmetic/long.h:57:22: note: expanded from macro 'RB_NUM2LONG'
#define RB_NUM2LONG  rb_num2long_inline   /**< @alias{rb_num2long_inline} */
                     ^
src/rbae.c:225:9: warning: 'AEUnflattenDesc' is deprecated: first deprecated in macOS 11.0 [-Wdeprecated-declarations]
  err = AEUnflattenDesc(RSTRING_PTR(data), &desc);
        ^~~~~~~~~~~~~~~
        AEUnflattenDescFromBytes
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AEDataModel.h:2461:1: note: 'AEUnflattenDesc' has been explicitly marked deprecated here
AEUnflattenDesc(
^
src/rbae.c:248:41: warning: implicit conversion loses integer precision: 'Size' (aka 'long') to 'int' [-Wshorten-64-to-32]
                                INT2NUM(dataSize)
                                ~~~~~~~ ^~~~~~~~
src/rbae.c:328:18: warning: implicit conversion loses integer precision: 'long' to 'int' [-Wshorten-64-to-32]
  return INT2NUM(length);
         ~~~~~~~ ^~~~~~
src/rbae.c:493:9: warning: 'LSFindApplicationForInfo' is deprecated: first deprecated in macOS 10.10 - Use LSCopyApplicationURLsForBundleIdentifier instead. [-Wdeprecated-declarations]
  err = LSFindApplicationForInfo(inCreator,
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/LSInfoDeprecated.h:765:1: note: 'LSFindApplicationForInfo' has been explicitly marked deprecated here
LSFindApplicationForInfo(
^
src/rbae.c:501:9: warning: 'FSRefMakePath' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
  err = FSRefMakePath(&outAppRef, path, PATH_MAX);
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:4086:18: note: 'FSRefMakePath' has been explicitly marked deprecated here
extern OSStatus  FSRefMakePath(const FSRef *ref, UInt8 *path, UInt32 pathBufferSize)     __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
                 ^
src/rbae.c:514:9: warning: 'FSPathMakeRef' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
  err = FSPathMakeRef((UInt8 *)StringValuePtr(path), &appRef, NULL);
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:4115:18: note: 'FSPathMakeRef' has been explicitly marked deprecated here
extern OSStatus  FSPathMakeRef(const UInt8 *path, FSRef *ref, Boolean *isDirectory)        __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
                 ^
src/rbae.c:517:11: warning: 'GetNextProcess' is deprecated: first deprecated in macOS 10.9 [-Wdeprecated-declarations]
    err = GetNextProcess(&psn);
          ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/ApplicationServices.framework/Frameworks/HIServices.framework/Headers/Processes.h:375:1: note: 'GetNextProcess' has been explicitly marked deprecated here
GetNextProcess(ProcessSerialNumber * pPSN)                     AVAILABLE_MAC_OS_X_VERSION_10_0_AND_LATER_BUT_DEPRECATED_IN_MAC_OS_X_VERSION_10_9;
^
src/rbae.c:519:11: warning: 'GetProcessBundleLocation' is deprecated: first deprecated in macOS 10.9 [-Wdeprecated-declarations]
    err = GetProcessBundleLocation(&psn, &foundRef);
          ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/ApplicationServices.framework/Frameworks/HIServices.framework/Headers/Processes.h:662:1: note: 'GetProcessBundleLocation' has been explicitly marked deprecated here
GetProcessBundleLocation(
^
src/rbae.c:521:9: warning: 'FSCompareFSRefs' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
    if (FSCompareFSRefs(&appRef, &foundRef) == noErr)
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:1850:15: note: 'FSCompareFSRefs' has been explicitly marked deprecated here
extern OSErr  FSCompareFSRefs(const FSRef *ref1, const FSRef *ref2)                     __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
              ^
src/rbae.c:533:9: warning: 'GetProcessForPID' is deprecated: first deprecated in macOS 10.9 [-Wdeprecated-declarations]
  err = GetProcessForPID(NUM2INT(pid), &psn);
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/ApplicationServices.framework/Frameworks/HIServices.framework/Headers/Processes.h:762:1: note: 'GetProcessForPID' has been explicitly marked deprecated here
GetProcessForPID(
^
src/rbae.c:546:9: warning: 'FSPathMakeRef' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
  err = FSPathMakeRef((UInt8 *)StringValuePtr(path), &appRef, NULL);
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:4115:18: note: 'FSPathMakeRef' has been explicitly marked deprecated here
extern OSStatus  FSPathMakeRef(const UInt8 *path, FSRef *ref, Boolean *isDirectory)        __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
                 ^
src/rbae.c:548:3: warning: 'LSApplicationParameters' is deprecated: first deprecated in macOS 10.10 - Use NSWorkspace instead. [-Wdeprecated-declarations]
  LSApplicationParameters appParams = {0,
  ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/LSOpenDeprecated.h:153:3: note: 'LSApplicationParameters' has been explicitly marked deprecated here
} LSApplicationParameters API_DEPRECATED("Use NSWorkspace instead.", macos(10.4,10.10) ) API_UNAVAILABLE( ios, tvos, watchos );
  ^
src/rbae.c:553:9: warning: 'LSOpenApplication' is deprecated: first deprecated in macOS 10.10 - Use -[NSWorkspace launchApplicationAtURL:options:configuration:error:] instead. [-Wdeprecated-declarations]
  err = LSOpenApplication(&appParams, &psn);
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/LSOpenDeprecated.h:185:1: note: 'LSOpenApplication' has been explicitly marked deprecated here
LSOpenApplication(
^
src/rbae.c:709:9: warning: 'FSPathMakeRef' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
  err = FSPathMakeRef((UInt8 *)StringValuePtr(path), &fsRef, NULL);
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:4115:18: note: 'FSPathMakeRef' has been explicitly marked deprecated here
extern OSStatus  FSPathMakeRef(const UInt8 *path, FSRef *ref, Boolean *isDirectory)        __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
                 ^
src/rbae.c:969:3: error: incompatible function pointer types passing 'VALUE (VALUE, VALUE, VALUE, SRefCon)' (aka 'unsigned long (unsigned long, unsigned long, unsigned long, void *)') to parameter of type 'VALUE (*)(VALUE, VALUE, VALUE, VALUE)' (aka 'unsigned long (*)(unsigned long, unsigned long, unsigned long, unsigned long)') [-Wincompatible-function-pointer-types]
  rb_define_module_function(mAE, "install_event_handler", rbAE_AEInstallEventHandler, 3);
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/matthewkennedy/.asdf/installs/ruby/3.3.0/include/ruby-3.3.0/ruby/internal/anyargs.h:338:142: note: expanded from macro 'rb_define_module_function'
#define rb_define_module_function(mod, mid, func, arity)    RBIMPL_ANYARGS_DISPATCH_rb_define_module_function((arity), (func))((mod), (mid), (func), (arity))
                                                                                                                                             ^~~~~~
/Users/matthewkennedy/.asdf/installs/ruby/3.3.0/include/ruby-3.3.0/ruby/internal/anyargs.h:274:1: note: passing argument to parameter here
RBIMPL_ANYARGS_DECL(rb_define_module_function, VALUE, const char *)
^
/Users/matthewkennedy/.asdf/installs/ruby/3.3.0/include/ruby-3.3.0/ruby/internal/anyargs.h:258:72: note: expanded from macro 'RBIMPL_ANYARGS_DECL'
RBIMPL_ANYARGS_ATTRSET(sym) static void sym ## _03(__VA_ARGS__, VALUE(*)(VALUE, VALUE, VALUE, VALUE), int); \
                                                                       ^
src/rbae.c:973:3: error: incompatible function pointer types passing 'VALUE (VALUE, VALUE, VALUE, SRefCon)' (aka 'unsigned long (unsigned long, unsigned long, unsigned long, void *)') to parameter of type 'VALUE (*)(VALUE, VALUE, VALUE, VALUE)' (aka 'unsigned long (*)(unsigned long, unsigned long, unsigned long, unsigned long)') [-Wincompatible-function-pointer-types]
  rb_define_module_function(mAE, "install_coercion_handler", rbAE_AEInstallCoercionHandler, 3);
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/matthewkennedy/.asdf/installs/ruby/3.3.0/include/ruby-3.3.0/ruby/internal/anyargs.h:338:142: note: expanded from macro 'rb_define_module_function'
#define rb_define_module_function(mod, mid, func, arity)    RBIMPL_ANYARGS_DISPATCH_rb_define_module_function((arity), (func))((mod), (mid), (func), (arity))
                                                                                                                                             ^~~~~~
/Users/matthewkennedy/.asdf/installs/ruby/3.3.0/include/ruby-3.3.0/ruby/internal/anyargs.h:274:1: note: passing argument to parameter here
RBIMPL_ANYARGS_DECL(rb_define_module_function, VALUE, const char *)
^
/Users/matthewkennedy/.asdf/installs/ruby/3.3.0/include/ruby-3.3.0/ruby/internal/anyargs.h:258:72: note: expanded from macro 'RBIMPL_ANYARGS_DECL'
RBIMPL_ANYARGS_ATTRSET(sym) static void sym ## _03(__VA_ARGS__, VALUE(*)(VALUE, VALUE, VALUE, VALUE), int); \
                                                                       ^
16 warnings and 2 errors generated.
make: *** [rbae.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/matthewkennedy/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/rb-scpt-1.0.3 for inspection.
Results logged to /Users/matthewkennedy/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/extensions/arm64-darwin-23/3.3.0/rb-scpt-1.0.3/gem_make.out
```

**Comment by zenspider** [comment]
I believe this is addressed by #23.

**Comment by zenspider** [comment]
Nevermind! that branch was actually merged to master about 2 months ago.... there's some confusing branch setup that needs to be resolved.

I think this project needs a little further cleanup, but then needs a gem release to resolve this issue.

**Comment by zenspider** [comment]
Alternatively, you should be able to check this out and create and install the gem yourself:

```
636 $ rake gem
rm -f rb-scpt-1.0.4.gem
  Successfully built RubyGem
  Name: rb-scpt
  Version: 1.0.4
  File: rb-scpt-1.0.4.gem
637 $ gem i ./rb-scpt-1.0.4.gem 
Building native extensions. This could take a while...
Successfully installed rb-scpt-1.0.4
Parsing documentation for rb-scpt-1.0.4
Installing ri documentation for rb-scpt-1.0.4
Done installing documentation for rb-scpt after 0 seconds
1 gem installed
```

**Comment by MatthewKennedy** [comment]
@zenspider The last release to hit Ruby Gems was July 13, 2018.

I don't know what branch was used when that gem was cut.

**Comment by MatthewKennedy** [comment]
I built and installed that version locally and it worked for me:

```bash
gem install rb-scpt-1.0.4.gem
Building native extensions. This could take a while...
Successfully installed rb-scpt-1.0.4
Building YARD (yri) index for rb-scpt-1.0.4...
Done installing documentation for rb-scpt after 0 seconds
1 gem installed
```

**Comment by zenspider** [comment]
After the last ruby update I had some issues with my private gem install. I couldn't figure out WHY but rubygems and/or rake-compiler stopped being happy building and it was installing the bundle file in the wrong location... I eventually bit the bullet and started a case sensitive partition and moved the code there and flushed out some more issues. I believe #29 solves the rest of these problems (at least, I am able to install it privately and use it w/o issues again).

I have gem release rights now so once that is merged I will work on getting a gem release out.

**Comment by zenspider** [comment]
Released v1.0.4!
